### DISCUSSION TITLE
Is there a plan/estimation to update the @walletconnect to use the v3 modal?

### DISCUSSION THREAD

**Discussion by kusky** [WalletConnect/walletconnect-monorepo#3884] [Developer Support] [answered]
I was implementing a solution using the walletconnect v2 for one of our websites using the modal and the provider, depending on some cases. But now that v3 is out, I have been tinkering with it and getting used to it (congrats btw, nice product), and the modal I have it working fine, but when I try to use the provider from @walletconnect/ethereum-provider it still triggers the v2 modal. Not sure if I am doing something wrong or it is just that it is an upcoming update (I checked the laboratories that you have and all of them are still using the v2 of the modal too related to the providers)
Thanks

**Comment by glitch-txs** [comment, marked as answer]
You should flag the connector as showQrModal: false

v3 refers to the Web3Modal SDK,
While it uses WalletConnect protocol v2

**Reply by kusky** [reply]
Ngl, the answer has passed above my head a bit, since now I don't trigger any interaction anymore and provider.connect() doesn't seem to do/open anything. But I guess I'm doing something wrong. I'll try to investigate further to understand how this would work, thanks

**Reply by glitch-txs** [reply]
ahh got you, the modal is just a UI design, I'm not sure if we'll update it tbh, I'd suggest using Web3Modal SDK instead though

**Reply by kusky** [reply]
Maybe I'm overcomplicating what I am trying to do. Our situation is, we work in L2 with Immutable X and they haven't adapted their api to your v2 even, so since you guys deprecated v1, we stopped having support for connecting wallets in mobile (which is a big part of our product since a big part of our community uses Metamask). So I've been working on making our own solution mixing your api with their provider object and our own wrappers (which is turning into a pain in the ass, truth be told). What I was trying to achieve was use your modal sdk as a provider and signer in L1, then wrap their provider around yours and then create my own class to be able to trigger it. 
But that QR modal part was throwing me off since, if it is false, I can't seem to find a way of getting anything back. And if I have it true, I saw the v2 of the modal while in the rest of my code I'm using v3. But if you say is purely a UI thing, I'll just use it as is. 

I think I've managed to make it work though, so now I can connect and sign with your provider and trigger registration on theirs with the same element. One think that I haven't managed to do is make your w3m components to refresh their data on screen after connecting through the ethereum provider. If I refresh the screen, they then work fine. But I can't find how to force that refresh by hand without a whole url hard refresh

**Reply by glitch-txs** [reply]
Why don't you use Web3Modal?
which library are you using? or just the raw provider

**Reply by kusky** [reply]
This is their new sdk library (they changed it not too long ago) https://www.npmjs.com/package/@imtbl/sdk
I can use your Web3Modal to do interactions to L1, but to interact with their L2 (registering a user, buying an order, depositing funds, etc) I need to use their API. That's why, as far as I could see, this way of using your provider to mix it with their and then make a wrapper myself is the only way I can achieve that. I could be wrong though

**Reply by glitch-txs** [reply]
Seems like it's not supported nor registered under the CAIP registry. You'll need to do the onboarding process

https://docs.walletconnect.com/advanced/multichain/chain-list

**Reply by glitch-txs** [reply]
You can use the universal provider for non-EVM chains, also you need to make sure there are wallets that support the chain with WalletConnect v2 integrated

https://docs.walletconnect.com/advanced/providers/universal
